pkg/controller/instanceset/instance_util.go

```diff
@@ -586,12 +586,12 @@ func buildInstanceTemplateExt(template workloads.InstanceTemplate, templateExt *
 		replicas = *template.Replicas
 	}
 	templateExt.Replicas = replicas
-	if template.NodeName != nil {
-		templateExt.Spec.NodeName = *template.NodeName
+	if template.SchedulingPolicy.NodeName != "" {
```

Comment:
do nil check of `SchedulingPolicy`

Reply:
<img width="626" alt="image" src="https://github.com/apecloud/kubeblocks/assets/31209634/3aa81172-81d9-4b6d-8f68-24d7a1ecdf04">

The `SchedulingPolicy` is not a pointer here.

Reply:
Indeed `SchedulingPolicy` should be optional. The pointer type is more appropriate. I will modify it.

Reply:
@free6om PTAL again, THX